pkg/kratos/service.go

```diff
@@ -657,6 +696,47 @@ func (s *Service) HasTOTPAvailable(ctx context.Context, id string) (bool, error)
 	return ok, nil
 }
 
+func (s *Service) HasNotEnoughLookupSecretsLeft(ctx context.Context, id string) (bool, error) {
+
+	identity, _, err := s.kratosAdmin.IdentityApi().
+		GetIdentity(ctx, id).
+		IncludeCredential([]string{"lookup_secret"}).
+		Execute()
+
+	if err != nil {
+		return false, err
+	}
+
+	lookupCredentials := identity.GetCredentials()["lookup_secret"].Config["recovery_codes"]
```

Comment:
possibile issue: do we know for certain that `"recovery_codes"` key will always be available?
If that's not the case, I'll put my suggestion here:

```go
lookUpSecret, ok := identity.GetCredentials()["lookup_secret"]
if !ok {
    // handle case key is not present
}

lookupCredentials, ok := lookUpSecret.Config["recovery_codes"]
if !ok {
    // handle case key is not present
}
```

Reply:
Technically yes, because this check is only executed if the user logged in with `lookup_secret` method, meaning the identity has recovery codes. But I added the additional checks in case anything goes wrong.